### PR TITLE
#441 [session03] 안가은 딥러닝2 과제, 딥러닝3 사전과제 제출합니다.

### PR DESCRIPTION
**Pull request by peanuut24**
### 🔍 어려웠던 점 및 새로 알게 된 점 간단히 정리
모델에 early stopping을 적용해도 에포크가 끝까지 돌아가는 경우가 있어서 신기했습니다. 그러나 epoch마다 val_loss가 꾸준히 나아져서 계속 학습을 했다기 보다는 patience가 조금 관대하게 설정되어 있어서 그랬다는 생각이 듭니다. 다음에 딥러닝 모델을 구현할때에 이 점을 신경써야겠다고 느꼈습니다!

### 🔍 질문 및 하고 싶은 말
세션 내용에 흥미를 자극하는 예시가 있어 재밌게 잘 들었습니다!!

---
### ✅ PR Checklist
PR이 다음 요구 사항을 충족하는지 확인하세요.
- [X] 본인 세션별 브랜치 → develop 브랜치 로 PR 합니다.

### CODE REVIEW

**Comment by yseo17** [comment]
과제와 사전과제 모두 꼼꼼하게 잘 해주셨습니다! 

얼리스타핑 부분을 유심히 잘 봐주신 것 같아요~~
early stopping 은 validation loss/accuracy를 기준으로 
일정 에폭(patience)동안 성능이 개선되지 않으면 학습이 중단되는 구조를 가지고 있는데요,

이 patience 값이 너무 크면 실제로 과적합이 되고 있음에도 조기 종료가 안될 수 있어요!
일반적으로는 3\~5 사이로 설정하지만 모델마다, 데이터 사이즈마다 적절한 값이 다를 수 있습니다~!

과제 하느라 고생하셨어요~~
